### PR TITLE
Implement IsAnyOfTheRolesAssignedToUser function

### PR DESCRIPTION
**Pull request by gayashanbc**
Resolves https://github.com/wso2/product-is/issues/6729

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -98,7 +98,7 @@
             <dependency>
                 <groupId>org.wso2.carbon</groupId>
                 <artifactId>org.wso2.carbon.logging</artifactId>
-                <version>${carbon.kernel.version}</version>
+                <version>${commons-logging.version}</version>
```
Comment (senthalan) [inline pom.xml]: We can't use the old carbon version.

File: pom.xml [hunk 2]
```diff
@@ -442,12 +447,13 @@
         </plugins>
     </build>
     <properties>
-        <carbon.kernel.version>4.4.32</carbon.kernel.version>
+        <carbon.kernel.version>4.6.0-alpha3</carbon.kernel.version>
```
Comment (senthalan) [inline pom.xml]: We need to update the relevant import versions of the kernel in all the components.

**Comment by senthalan** [comment]
Please note that, as this change include a kernel minor version update. We need to bump a minor version of this repository when merging this PR.

File: components/org.wso2.carbon.identity.conditional.auth.functions.user/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/user/IsAnyOfTheRolesAssignedToUserFunctionImpl.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        String tenantDomain = user.getTenantDomain();
+        String userStoreDomain = user.getUserStoreDomain();
+        String username = user.getUserName();
+        try {
+            UserRealm userRealm = Utils.getUserRealm(user.getTenantDomain());
+            if (userRealm != null) {
+                return Utils.getUserStoreManager(tenantDomain, userRealm, userStoreDomain);
+            }
+        } catch (FrameworkException e) {
+            LOG.error("Error in evaluating the function ", e);
```
Comment (senthalan) [inline components/org.wso2.carbon.identity.conditional.auth.functions.user/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/user/IsAnyOfTheRolesAssignedToUserFunctionImpl.java]: Let change the error message to specify error during getting the userstore

**Comment by senthalan** [comment]
With the kernel update, it is better if we remove the scr plugin as well.
https://github.com/wso2-extensions/identity-outbound-auth-requestpath-oauth/pull/21/commits/e362d0c37065c23bbb9c093e50f9a46da0d949c8

File: pom.xml (hunk 2 above)
Comment (gayashanbc) [reply pom.xml]: Resolved in https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/53/commits/99c9587ba62eccd099e371f7cda4cec4aa8c649b

**Pull request merged**